src/context/TrashProvider.js

```diff
@@ -0,0 +1,75 @@
+import React, { createContext, useContext, useState } from "react";
+
+import {
+  collection,
+  doc,
+  deleteDoc,
+  addDoc,
+  serverTimestamp,
+} from "firebase/firestore";
+import { db } from "../firebase.config";
+
+const TrashContext = createContext();
+
+export const TrashProvider = ({ children }) => {
+  const [trashedNotes, setTrashedNotes] = useState([]);
+
+  const deleteNote = (note) => {
+    const docRef = doc(db, "notes", note.id);
+    const colRef = collection(db, "trash");
+    addDoc(colRef, {
+      archived: false,
+      body: note.body,
+      color: note.color,
+      createdAt: note.createdAt,
+      pinned: note.pinned,
+      tags: note.tags,
+      title: note.title,
+      trashed: true,
+      updatedAt: note.updatedAt,
+      trashedAt: serverTimestamp(),
+      id: note.id,
+    });
+    deleteDoc(docRef);
+  };
+
+  const restoreNote = (note) => {
+    console.log(note);
```

Comment:
Is this console statement required? If not then I think, you can remove this.

Reply:
Removing it.